runtime/Java/src/org/antlr/v4/runtime/atn/ATNDeserializer.java

```diff
@@ -20,10 +20,10 @@
  * @author Sam Harwell
  */
 public class ATNDeserializer {
-	public static final int SERIALIZED_VERSION;
-	static {
-		SERIALIZED_VERSION = 4;
-	}
+
+  static final int LEGACY_SERIALIZED_VERSION = 3;
```

Comment:
It's unclear what exacly `LEGACY_SERIALIZED_VERSION` is. What if `SERIALIZED_VERSION` will be incremented?

Reply:
This is a loose patch to solve an internal problem we have.